spec/Section 3 -- Type System.md

```diff
@@ -2168,3 +2168,150 @@ to the relevant IETF specification.
 ```graphql example
 scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
 ```
+
+## Schema Coordinates
+
+SchemaCoordinate :
+
+- TypeCoordinate
+- MemberCoordinate
+- ArgumentCoordinate
+- DirectiveCoordinate
+- DirectiveArgumentCoordinate
+
+TypeCoordinate : Name
+
+MemberCoordinate : Name . Name
+
+ArgumentCoordinate : Name . Name ( Name : )
+
+DirectiveCoordinate : @ Name
+
+DirectiveArgumentCoordinate : @ Name ( Name : )
+
+:: A _schema coordinate_ is a human readable string that uniquely identifies a
+_schema element_ within a GraphQL Schema.
+
+:: A _schema element_ can be a named type, a field, an input field, an enum
+value, a field argument, a directive, or a directive argument.
+
+A _schema coordinate_ is always unique. Each _schema element_ can be referenced
+by exactly one possible schema coordinate.
+
+A _schema coordinate_ may refer to either a defined or built-in _schema
+element_. For example, `String` and `@deprecated(reason:)` are both valid schema
+coordinates which refer to built-in schema elements. However it must not refer
+to a meta-field. For example, `Business.__typename` is _not_ a valid schema
+coordinate.
+
+Note: Union members are not valid _schema coordinate_ as they reference existing
+types in the schema. This preserves the uniqueness property of a _schema
+coordinate_ as stated above.
+
+Note: A {SchemaCoordinate} is not a definition within a GraphQL {Document}, but
+a separate standalone grammar, intended to be used by tools to reference types,
+fields, and other _schema element_. Examples include: references within
+documentation to refer to types and fields in a schema, a lookup key that can be
+used in logging tools to track how often particular fields are queried in
+production.
+
+**Resolving a Schema Coordinate**
+
+To refer to a _schema element_, a _schema coordinate_ must be interpreted in the
+context of a GraphQL {schema}.
+
+If the _schema element_ cannot be found, the resolve function will not yield a
+value (without raising an error). However, an error will be raised if any
+non-leaf nodes within a _schema coordinate_ cannot be found in the {schema}.
+
+TypeCoordinate : Name
+
+1. Let {typeName} be the value of {Name}.
+2. Return the type in the {schema} named {typeName}, or {null} if no such type
+   exists.
+
+MemberCoordinate : Name . Name
+
+1. Let {typeName} be the value of the first {Name}.
```

Comment:
```suggestion
1. Let {typeName} be the value of the first {Name}.
2. Assert: {typeName} must not be a meta-field.
```

Reply:
Teeing this up in anticipation that we decide on this for now.

Could also be phrased as `Assert: {typeName} must not begin with {__} (two underscores)`

Given this:

> All types and directives defined within a schema must not have a name which begins with "__" (two underscores), as this is used exclusively by GraphQL’s introspection system.

but former seems preferred for brevity?

(that is, assuming that all fields prefixed with `__` are meta-fields and vice versa. I think this is the case but not explicitly stated this is the same set afaict?)